### PR TITLE
Improve test selection for TF unittest runner

### PR DESCRIPTION
**Pull request by bani-intelaipg**
- handle exact test name match
- The later RUN inavlidates previous SKIP, and vice-versa

### CODE REVIEW

File: test/python/tensorflow/tf_unittest_runner.py [hunk 1]
```diff
@@ -243,10 +244,10 @@ def list_tests(module_list, regex_input):
                 alltests.append(aTestCase.id())
 
     for aTestCaseID in alltests:
```
Comment (adk9) [inline test/python/tensorflow/tf_unittest_runner.py:246]: Can you please follow the style here: https://google.github.io/styleguide/pyguide.html?

File: test/python/tensorflow/tf_unittest_runner.py [hunk 2]
```diff
@@ -267,7 +268,13 @@ def read_tests_from_manifest(manifestfile, tensorflow_path):
     Reads a file that has include & exclude patterns,
     Returns a list of leaf-level single testcase, no duplicates
     """
-    list_of_tests = []
+
+    def invalidate_A_after_B(AList, BList):
+        for aTest in BList:
+            if aTest in AList:
+                AList.remove(aTest)
+
```
Comment (adk9) [inline test/python/tensorflow/tf_unittest_runner.py:276]: Use "sets" for tests and you can simply take a set difference for this. You'll also be able to eliminate any duplicated tests as a result of using sets.

**Review by adk9 — CHANGES_REQUESTED** [review]

File: test/python/tensorflow/tf_unittest_runner.py [hunk 3]
```diff
@@ -243,10 +244,10 @@ def list_tests(module_list, regex_input):
                 alltests.append(aTestCase.id())
 
     for aTestCaseID in alltests:
-        if regex_input in aTestCaseID:  # substring match
+        if regex_input == aTestCaseID:  # exact match
             listtests.append(aTestCaseID)
-        elif regex_input.count('*') > 0:
-            regex_pattern = regex_input
+        else:
+            regex_pattern = '^' + regex_input + '$'
```
Comment (bani-intelaipg) [inline test/python/tensorflow/tf_unittest_runner.py:250]: This was buggy earlier.

**Review by adk9 — APPROVED** [review]
Looks good, thanks.

**Pull request merged**